src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -124,6 +124,17 @@ public WorkspaceStage getWorkspaceStage(UUID workspaceId) {
     return WorkspaceStage.valueOf(jdbcTemplate.queryForObject(sql, params, String.class));
   }
 
+  // TODO: do all three variants of assertMcWorkspace belong in WorkspaceDAO?
+  private void assertMcWorkspace(WorkspaceStage stage, UUID workspaceId, String action) {
```

Comment:
nit: It would be nice to make the actions an enum with a string description constant, but that can be another PR.

Reply:
instead of making this an enum, I have clarified its usage in the javadoc, and renamed the argument to reflect its true usage. Does that work?

Reply:
sure. we could change later if we find we want a pie chart of actions or something.